ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -1173,9 +1259,14 @@ void maybeNotifyForChunkCreation(PutChunk chunk) {
      * blob is composite. If no first chunk was put successfully, this will do nothing.
      */
     void notifyForFirstChunkCreation() {
-      String chunkId = firstChunkIdAndProperties.getFirst().getID();
-      BlobProperties chunkProperties = firstChunkIdAndProperties.getSecond();
-      notificationSystem.onBlobCreated(chunkId, chunkProperties, userMetadata, NotificationBlobType.DataChunk);
+      if (firstChunkIdAndProperties != null) {
+        // reason to check for not null: with encryption in play, there are chances that 2nd chunk's encrypt job callback
```

Comment:
is this because of encryption only? Without encryption can this happen ? for e.g. if the second chunk's put succeeded but not the first?

Reply:
yeah. It could happen I guess. @pnarayanan : can you comment on this

Reply:
This method is called on first chunk creation only - so I don't think these comments apply - please also verify if you are doing the right thing (I will take a better look later).